File: app/src/main/java/ir/ghararemaghzha/game/adapters/BuyHistoryAdapter.java
package ir.ghararemaghzha.game.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.textview.MaterialTextView;

import java.util.List;

import ir.ghararemaghzha.game.R;
import ir.ghararemaghzha.game.classes.DateConverter;
import ir.ghararemaghzha.game.classes.Utils;
import ir.ghararemaghzha.game.models.BuyHistoryModel;

public class BuyHistoryAdapter extends RecyclerView.Adapter<BuyHistoryAdapter.ViewHolder> {

    private List<BuyHistoryModel> data;
    private Context context;

    public BuyHistoryAdapter(Context context, List<BuyHistoryModel> data) {
        this.data = data;
        this.context = context;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.row_buyhistory, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder h, int position) {
        BuyHistoryModel model = data.get(position);
        if (model != null) {
            h.title.setText(context.getString(R.string.buyhistory_row_title, model.getPlan()+"000"));

            h.price.setText(context.getString(R.string.buyhistory_row_price, Utils.moneySeparator(model.getAmount())));
            if (model.getInfluencerCode() != null && !model.getInfluencerCode().isEmpty()) {
                h.influencer.setVisibility(View.VISIBLE);
                h.influencer.setText(context.getString(R.string.buyhistory_row_price, Utils.moneySeparator(model.getAmount())));
            } else
                h.influencer.setVisibility(View.GONE);

            if (model.getUserId().equals("1")) {
                h.stat.setText(context.getString(R.string.buyhistory_row_stat, "موفق"));
                h.resCode.setVisibility(View.VISIBLE);
                h.resCode.setText(context.getString(R.string.buyhistory_row_rescode, model.getResCode()));
            } else {
                h.stat.setText(context.getString(R.string.buyhistory_row_stat, "ناموفق"));
                h.resCode.setVisibility(View.GONE);


            }

            String date = model.getDate();
            DateConverter dateConverter = new DateConverter();

            dateConverter.gregorianToPersian(Integer.parseInt(date.substring(0, 4)), Integer.parseInt(date.substring(5, 7)), Integer.parseInt(date.substring(8, 10)));
            h.date.setText(context.getString(R.string.buyhistory_row_date, date.substring(11, 16)+ " " + dateConverter.getYear() + "/" + dateConverter.getMonth() + "/" + dateConverter.getDay() ));


        }

    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        private MaterialTextView title, price, date, influencer, stat, resCode;

        public ViewHolder(@NonNull View v) {
            super(v);
            title = v.findViewById(R.id.buyhistory_row_description);
            price = v.findViewById(R.id.buyhistory_row_price);
            date = v.findViewById(R.id.buyhistory_row_date);
            influencer = v.findViewById(R.id.buyhistory_row_influencer);
            stat = v.findViewById(R.id.buyhistory_row_stat);
            resCode = v.findViewById(R.id.buyhistory_row_rescode);

        }
    }
}

File: app/src/main/java/ir/ghararemaghzha/game/adapters/BuyHistoryAdapter.kt
package ir.ghararemaghzha.game.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.textview.MaterialTextView
import ir.ghararemaghzha.game.R
import ir.ghararemaghzha.game.classes.DateConverter
import ir.ghararemaghzha.game.classes.Utils
import ir.ghararemaghzha.game.models.BuyHistoryModel

class BuyHistoryAdapter(private val ctx: Context,
                        private val data: List<BuyHistoryModel>) : RecyclerView.Adapter<BuyHistoryAdapter.ViewHolder>() {

    class ViewHolder(v: View) :RecyclerView.ViewHolder(v){
        val title:MaterialTextView = v.findViewById(R.id.buyhistory_row_description)
        val price:MaterialTextView = v.findViewById(R.id.buyhistory_row_price)
        val date:MaterialTextView = v.findViewById(R.id.buyhistory_row_date)
        val influencer:MaterialTextView = v.findViewById(R.id.buyhistory_row_influencer)
        val stat:MaterialTextView = v.findViewById(R.id.buyhistory_row_stat)
        val resCode:MaterialTextView = v.findViewById(R.id.buyhistory_row_rescode)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder=ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.row_buyhistory, parent, false))

    override fun onBindViewHolder(h: ViewHolder, position: Int) {
        val model = data[position]
        when (model.plan) {
            "1" -> h.title.text = ctx.getString(R.string.buyhistory_row_title, "500")
            "2" -> h.title.text = ctx.getString(R.string.buyhistory_row_title, "1000")
            "3" -> h.title.text = ctx.getString(R.string.buyhistory_row_title, "1500")
            "4" -> h.title.text = ctx.getString(R.string.buyhistory_row_title, "2000")
            "5" -> h.title.text = ctx.getString(R.string.buyhistory_row_title, "2500")
            else -> h.title.text = ctx.getString(R.string.buyhistory_row_title2)
        }
        h.price.text = ctx.getString(R.string.buyhistory_row_price, Utils.moneySeparator(model.amount))
        if (model.influencerCode != null && model.influencerCode.isNotEmpty()) {
            h.influencer.visibility = View.VISIBLE
            h.influencer.text = ctx.getString(R.string.buyhistory_influencer_price, Utils.moneySeparator(model.amount))
        } else h.influencer.visibility = View.GONE
        if (model.valid == "1") {
            h.stat.text = ctx.getString(R.string.buyhistory_row_stat, "موفق")
            h.resCode.visibility = View.VISIBLE
            h.resCode.text = ctx.getString(R.string.buyhistory_row_rescode, model.resCode)
        } else {
            h.stat.text = ctx.getString(R.string.buyhistory_row_stat, "ناموفق")
            h.resCode.visibility = View.GONE
        }
        val date = model.date
        val dateConverter = DateConverter()
        dateConverter.gregorianToPersian(date.substring(0, 4).toInt(), date.substring(5, 7).toInt(), date.substring(8, 10).toInt())
        h.date.text = ctx.getString(R.string.buyhistory_row_date, date.substring(11, 16) + " " + dateConverter.year + "/" + dateConverter.month + "/" + dateConverter.day)

    }

    override fun getItemCount(): Int =data.size
}